socorro/app/socorro_app.py

```diff
@@ -362,6 +362,61 @@ def setup_logger(config, local_unused, args_unused):
     return wrapped_logger
 
 
+def setup_metrics(config, local_unused, args_unused):
+    """Sets up metrics client which is either a DogStatsd client or a LoggingClient
+
+    NOTE(willkg): This is a little gross, but that's mostly to keep all the code in one place and
+    minimal which makes it easier to deal with for now. If this ever grows, then we'll probably want
+    to split this out into its own module.
+
+    This gets tossed in config (along with everything else), so you can use it like this::
+
+        class SomeComponent(RequiredConfig):
+            def something(self):
+                self.config.metrics.increment('somekey')
+
+    """
+    # We do the import here in case there are parts of Socorro that are running in environments that
+    # don't have this library installed.
+    #
+    # FIXME(willkg): We're using a *super* old version of dogstatsd-python which doesn't have the
+    # DogStatsd class. We should upgrade.
+    try:
+        from datadog.dogstatsd import statsd
+    except ImportError as ioe:
+        statsd is None
```

Comment:
I have no idea if there exist some things that use `SocorroApp` or some subclass that run in an environment that doesn't have the dogstatsd-python library. So I did this goofy thing.

Amongst other things, it makes it easier to deal with in the tests since many tests create configuration environments, so because of how this is structured, all those tests use `LoggingMetrics` without any adjustment. Yay!